cornflow/endpoints/execution.py

```diff
@@ -171,17 +172,28 @@ def get(self, idx):
 
         self.user_id, self.admin, self.super_admin = Auth.return_user_info(request)
         execution = ExecutionModel.get_one_execution_from_user(self.user_id, idx)
-        if execution.finished:
+        if execution.state == 1:
             return {'status': 'finished'}, 200
         dag_run_id = execution.dag_run_id
         if not dag_run_id:
+            execution.state = -1
+            execution.state_message = 'The execution has no dag_run associated'
+            execution.save()
             return {'error': 'The execution has no dag_run associated'}, 400
         af_client = Airflow(**airflow_conf)
         if not af_client.is_alive():
+            execution.state = -1
```

Comment:
maybe this one is the same as ERROR_START. Ideally, if the state_message changes, the error should be different.